### DISCUSSION TITLE
What changed in 1.7.0 that affected package installs?

### DISCUSSION THREAD

**Discussion by JaimeHolland** [python-poetry/poetry#9531] [General] [answered]
Hi, I am using Poetry as a dependency manager only - one of the packages I want to install is amazon_kclpy.  [This package has a setup.py file](https://github.com/awslabs/amazon-kinesis-client-python/blob/master/setup.py) that downloads some Java JARs that are required for the functioning of the application.

This works in Poetry 1.6.1, but does not work in 1.7.0 - the package is installed without errors, but the JARs are not downloaded.

[The Poetry release notes ](https://python-poetry.org/history/#170---2023-11-03)indicate changes around `setuptools` and `wheel`... is it something around this?  I would like to be able to use a more recent version of Poetry, and retain the functionality of the JARs being downloaded

I've put this in docker so it can be reproduced:

```
FROM python:3.11-slim-bookworm

RUN pip install pip==24.1.1

# Does download JARs
RUN pip install poetry==1.6.1 

# Doesn't download JARs
# RUN pip install poetry==1.7.0 

WORKDIR /app

RUN poetry init --no-interaction
RUN poetry add amazon_kclpy==2.1.5

# Print out to a local file whether the JARs have been downloaded
RUN poetry run python -c "import amazon_kclpy; import os; print('\n'.join(os.listdir(os.path.join(os.path.dirname(amazon_kclpy.__file__), 'jars'))))" > output.txt

```

**Comment by dimbleby** [comment, marked as answer]
updated https://pypi.org/project/build/ probably
